fulp_modules/features/antagonists/bloodsuckers/code/powers/brujah/brash.dm

```diff
@@ -0,0 +1,152 @@
+		if(level_current < 4 || (istype(W, /obj/structure/window/reinforced) && level_current < 5))
+			target_atom.balloon_alert(owner, "you need more ranks to destroy that!")
+			return
+		if(istype(W, /obj/structure/window/reinforced) || istype(W, /obj/structure/window/plasma))
+			HitWithStyle(W, 'sound/effects/bang.ogg', 30, 25, 15 SECONDS)
+		else
+			HitWithStyle(W, 'sound/effects/bang.ogg', 20, 15, 10 SECONDS)
+		return
+
+	if(istype(target_atom, /obj/machinery/door/window))
+		HitWithStyle(target_atom, 'sound/effects/bang.ogg', 50, 10, 5 SECONDS)
+		return
+
+	if(istype(target_atom, /obj/structure/table))
+		HitWithStyle(target_atom, 'sound/effects/bang.ogg', 35, 10, 5 SECONDS)
+
+	if(!iswallturf(target_atom))
+		return
+	if(level_current < 4 || (istype(target_atom, /turf/closed/wall/r_wall) && level_current < 5))
+		target_atom.balloon_alert(owner, "you need more ranks to destroy that!")
+		return
+	if(isindestructiblewall(target_atom))
+		target_atom.balloon_alert(owner, "this wall is indestructible!")
+		return
+
+	/// If we get past all of the if statements then it's almost certainly a wall at this point.
+	rip_and_tear(owner, target_atom)
+
+/// Copied over from '/datum/element/wall_tearer/proc/rip_and_tear' with appropriate adjustment.
+/datum/action/cooldown/bloodsucker/targeted/brawn/brash/proc/rip_and_tear(mob/living/tearer, atom/target)
+	var/tear_time = 0.75 SECONDS
+	var/reinforced_multiplier = 5
+	var/rip_time = (istype(target, /turf/closed/wall/r_wall) ? tear_time * reinforced_multiplier : tear_time)
+
+	if(istype(target, /turf/closed/wall/r_wall))
+		bloodcost = 40
+		cooldown_time = 20 SECONDS
+	else
+		bloodcost = 20
+		cooldown_time = 15 SECONDS
+
+	while(istype(target, /turf/closed/wall))
+		tearer.visible_message(span_warning("[tearer] viciously rips into [target]!"))
+		playsound(tearer, 'sound/machines/airlock_alien_prying.ogg', vol = 50, vary = TRUE, frequency = 2)
+		target.balloon_alert(tearer, "tearing...")
+
+		if (!do_after(tearer, delay = rip_time, target = target, interaction_key = "bloodsucker interaction"))
+			tearer.balloon_alert(tearer, "interrupted!")
+			return
+
+		tearer.do_attack_animation(target)
+		target.AddComponent(/datum/component/torn_wall)
+		tearer.UnarmedAttack(target, proximity_flag = TRUE)
+
+/// TODO: check if switch statements work with istype()
+/// and maybe refactor 'CheckValidTarget()'/'CheckCanTarget()' entirely
+/datum/action/cooldown/bloodsucker/targeted/brawn/brash/CheckValidTarget(atom/A)
+	if(A == owner)
+		return FALSE
+	if(isliving(A))
+		return TRUE
+	if(istype(A, /obj/machinery/door))
+		return TRUE
+	if(istype(A, /obj/machinery/door/window))
+		return TRUE
+	if(istype(A, /obj/structure/table) || istype(A, /obj/structure/table_frame))
+		return TRUE
+	if(istype(A, /obj/structure/closet))
+		return TRUE
+	if(istype(A, /obj/structure/girder))
+		return TRUE
+	if(istype(A, /obj/structure/grille) && !istype(A, /obj/structure/grille/indestructible))
```

Comment:
maybe we should just wholesale check for a ``INDESTRUCTIBLE`` resistance flag instead of hardcoded istype checks

Reply:
Done